packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts

```diff
@@ -45,32 +45,37 @@ export class IndexMetadataService {
 
     const indexName = `IDX_${generateDeterministicIndexName([tableName, ...columnNames])}`;
 
-    let savedIndexMetadata: IndexMetadataEntity;
+    let result: InsertResult;
 
     try {
-      savedIndexMetadata = await this.indexMetadataRepository.save({
-        name: indexName,
-        tableName,
-        indexFieldMetadatas: fieldMetadataToIndex.map(
-          (fieldMetadata, index) => {
-            return {
-              fieldMetadataId: fieldMetadata.id,
-              order: index,
-            };
-          },
-        ),
-        workspaceId,
-        objectMetadataId: objectMetadata.id,
-        ...(isDefined(indexType) ? { indexType: indexType } : {}),
-        isCustom: isCustom,
-      });
+      result = await this.indexMetadataRepository.upsert(
```

Comment:
Should we rename the method upsertIndex? @ijreilly

Reply:
I wouldn't - it still creates the index, the upsert only concerns the entry in metadata table. but a migration to create the index is added anyway

Reply:
You are right @ijreilly!